### DISCUSSION TITLE
Option to not change float precision?

### DISCUSSION THREAD

**Discussion by GwenNealon** [j-brooke/FracturedJson#12] [Ideas]
Hey! Thanks so much for making this, it works really well! My only concern with using it is the removal of precision for floating numbers since I am using this with Python and engineering parameters where that matters more. Would it be possible to add a "don't change decimal precision" option? Either way, thanks!

**Comment by j-brooke** [comment]
If you're using the .NET version of the code, your numbers will be left alone if you set the property `DontJustifyNumbers=true`.  In that case, it just copies the string form of the number from the source to the destination.  (If you find a case where it doesn't behave that way, please let me know.)

For the JavaScript version (and thus web and VSCode ones), numbers are unavoidably parsed as double precision floats before being written back out.  Addressing that will require rewriting using a JSON parsing library rather than relying on `JSON.parse`.  (I may do that at some point - lots of requested features depend on it - but it's no quick fix.)

There's a related discussion in [this issue](https://github.com/j-brooke/FracturedJson/issues/11).

**Reply by GwenNealon** [reply]
Got it! Thanks for the quick reply!

**Comment by j-brooke** [comment]
As of version 3, all implementations of FracturedJson correctly preserve numbers, no matter how many digits.
